### PR TITLE
Add a faq about using type hints for performance optimizations

### PR DESCRIPTION
**Pull request by giltayar**
Based on #75

### CODE REVIEW

**Review by ljharb — APPROVED** [review]
Love the wording here, I think it captures both important points

**Review by orta — APPROVED** [review]

**Review by bl0cknumber — APPROVED** [review]

**Review by romulocintra — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -714,6 +714,11 @@ That is one reason why this proposal endorses purely static types.
 
 If there is a desire to keep language open to later adding runtime-checked types, in addition to the static types proposed here, we could make an explicit syntax reservation in the grammar to support both.
 
+### Could this proposal enable runtimes to optimize performance based on type hints
```
Comment (robpalme) [inline README.md:717]:
```suggestion
### Could this proposal enable runtimes to optimize performance based on type hints?
```

File: README.md [hunk 2]
```diff
@@ -714,6 +714,11 @@ That is one reason why this proposal endorses purely static types.
...
+
+Given that the type syntax in the type hints are opaque to the runtimes, this proposal as it stands would not allow performance improvements.
+In any case, it is a non-goal of this proposal to improve the performance of JavaScript.
```
Comment (robpalme) [inline README.md:720]:
```suggestion
Whilst it is possible to theorize runtime optimizations driven by statically declared types, the proposal authors are not aware of successful experiments in JavaScript that meaningfully beat dynamic type-driven JIT optimization.

The proposed type syntax has no defined semantics and so is opaque to the runtimes.
It is equivalent to asking _"Could a runtime use `/** comments **/` to optimize performance?"_
To which the answer is:  almost certainly not - at least not in a standard way.
Therefore this proposal alone does not directly offer new opportunities for performance improvements.

It is explicitly a **non-goal** of this proposal to improve the performance of JavaScript.
```

**Review by robpalme — COMMENTED** [review]
Thanks for answering this!

**Review by romulocintra — APPROVED** [review]

**Review by robpalme — APPROVED** [review]

**Comment by ptomato** [comment]
Really nice and clear way of putting it, my compliments!

**Pull request merged**